server/migrations/2020-06-30-135809_remove_mat_views/up.sql

```diff
@@ -0,0 +1,576 @@
+returns trigger language plpgsql
+as $$
+begin
+  -- TODO possibly select from comment_fast to get previous scores, instead of re-fetching the views?
+  IF (TG_OP = 'DELETE') THEN
+    delete from comment_aggregates_fast where id = OLD.comment_id;
+    insert into comment_aggregates_fast select * from comment_aggregates_view where id = OLD.comment_id;
+  ELSIF (TG_OP = 'UPDATE') THEN
+    delete from comment_aggregates_fast where id = NEW.comment_id;
+    insert into comment_aggregates_fast select * from comment_aggregates_view where id = NEW.comment_id;
+  ELSIF (TG_OP = 'INSERT') THEN
+    delete from comment_aggregates_fast where id = NEW.comment_id;
+    insert into comment_aggregates_fast select * from comment_aggregates_view where id = NEW.comment_id;
+  END IF;
+
+  return null;
+end $$;
+
+drop trigger refresh_comment_like on comment_like;
+create trigger refresh_comment_like
+after insert or update or delete
+on comment_like
+for each row
+execute procedure refresh_comment_like();
+
+-- Community user ban
+
+drop trigger refresh_community_user_ban on community_user_ban;
+create trigger refresh_community_user_ban
+after insert or update or delete
+on community_user_ban
+for each row
+execute procedure refresh_community_user_ban();
+
+-- select creator_name, banned_from_community from comment_fast_view where user_id = 4 and content = 'test_before_ban';
+-- select creator_name, banned_from_community, community_id from comment_aggregates_fast where content = 'test_before_ban';
+-- Sample insert 
+-- insert into comment(creator_id, post_id, content) values (1198, 341, 'test_before_ban');
+-- insert into community_user_ban(community_id, user_id) values (2, 1198);
+-- Sample delete
+-- delete from community_user_ban where user_id = 1198 and community_id = 2;
+-- delete from comment where content = 'test_before_ban';
+-- update comment_aggregates_fast set banned_from_community = false where creator_id = 1198 and community_id = 2;
+create or replace function refresh_community_user_ban()
+returns trigger language plpgsql
+as $$
+begin
+  -- TODO possibly select from comment_fast to get previous scores, instead of re-fetching the views?
+  IF (TG_OP = 'DELETE') THEN
+    update comment_aggregates_fast set banned_from_community = false where creator_id = OLD.user_id and community_id = OLD.community_id;
+    update post_aggregates_fast set banned_from_community = false where creator_id = OLD.user_id and community_id = OLD.community_id;
+  ELSIF (TG_OP = 'INSERT') THEN
+    update comment_aggregates_fast set banned_from_community = true where creator_id = NEW.user_id and community_id = NEW.community_id;
+    update post_aggregates_fast set banned_from_community = true where creator_id = NEW.user_id and community_id = NEW.community_id;
+  END IF;
+
+  return null;
+end $$;
+
+-- Community follower
+
+drop trigger refresh_community_follower on community_follower;
+create trigger refresh_community_follower
+after insert or update or delete
```

Comment:
No need to have this fire on update (ditto for any case I missed)

Reply:
Ah good catch, thx.